vivisect/__init__.py

```diff
@@ -606,6 +611,9 @@ def analyze(self):
         Call this to ask any available analysis modules
         to do their thing...
         """
+        if self.getMeta('GUID') == None:
```

Comment:
We could probably manage both ( generate randomly on startup but let saved data override on loading a workspace ).  Trying to introduce the workspace guid (which will be super fundamental eventually i suspect) when some function or another is called seems to introduce a state machine for it being set or not :)